src/vsr/replica.zig

```diff
@@ -3148,8 +3148,10 @@ pub fn ReplicaType(
                     self.state_machine.checkpoint(commit_op_checkpoint_state_machine_callback);
                     self.client_sessions_checkpoint.checkpoint(commit_op_checkpoint_client_sessions_callback);
                     self.client_replies.checkpoint(commit_op_checkpoint_client_replies_callback);
+                    // The grid checkpoint must begin after the manifest/trailers have acquired all
+                    // their blocks, since it encodes the free set:
+                    self.grid.checkpoint(commit_op_checkpoint_grid_callback);
```

Comment:
If we do this _first_, is there any assert that can catch us? Basically, that no reservations can be created while the free set is being checkpointed?

I wonder if something like 

```diff
diff --git a/src/vsr/grid.zig b/src/vsr/grid.zig
index d86d33bf..3bc8660e 100644
--- a/src/vsr/grid.zig
+++ b/src/vsr/grid.zig
@@ -334,6 +334,7 @@ pub fn GridType(comptime Storage: type) type {
 
             {
                 assert(grid.free_set.count_reservations() == 0);
+                grid.free_set.opened = false;
 
                 grid.free_set.include_staging();
                 defer grid.free_set.exclude_staging();
@@ -361,6 +362,7 @@ pub fn GridType(comptime Storage: type) type {
         fn checkpoint_free_set_callback(set: *CheckpointTrailer) void {
             const grid = @fieldParentPtr(Grid, "free_set_checkpoint", set);
             assert(grid.callback == .checkpoint);
+            grid.free_set.opened = true;
             grid.checkpoint_join();
         }
 ```

could do the trick?

Reply:
This is already guarded against via the `grid.callback == .none` assertions in `Grid.{reserve,acquire}`.